lib/pbench/server/auth/__init__.py

```diff
@@ -116,15 +118,23 @@ def set_well_known_endpoints(self):
             raise
 
     def set_oidc_public_key(self):
-        """Sets the public key of the current OIDC client"""
+        """Sets the public key of the current OIDC client.
+
+        Raises:
+            KeyError: if public key not found in response payload
+            Requests session object Exceptions
+        """
 
         realm_public_key_uri = f"{self.server_url}/realms/{self.realm_name}"
-        public_key = (
-            "-----BEGIN PUBLIC KEY-----\n"
-            + self._get(realm_public_key_uri).json()["public_key"]
-            + "\n-----END PUBLIC KEY-----"
+        response_json = self._get(realm_public_key_uri).json()
+        try:
+            public_key = response_json["public_key"]
+        except KeyError:
+            self.logger.warning("OIDC Public key not found in {}", response_json)
```

Comment:
I don't think this is a "warning" by the loose conventions we'd discussed. That is, it's not a user-error, right? Does it need to be logged at all given that we're just raising it again? In fact, if the intent is to defer exception handling to the caller, why have a `try` at all?

Reply:
Yes, that's true. I added this block when Peter was suggesting some of these changes but I didn't realize I committed them. 

> intent is to defer exception handling to the caller, why have a `try` at all

Sure thing!